common/src/main/java/io/pravega/common/util/TypedProperties.java

```diff
@@ -155,9 +154,43 @@ public boolean getBoolean(Property<Boolean> property) throws ConfigurationExcept
      *                                does not have a default value set, or when the property cannot be parsed as a positive Integer.
      */
     public int getPositiveInt(Property<Integer> property) {
+        return getPositiveInt(property, true);
+    }
+
+    /**
+     * Gets the value of an Integer property only if it is greater than 0.
+     *
+     * @param property The Property to get.
+     * @param strict   If true, then the value must be strictly greater than 0; otherwise 0 is accepted.
+     * @return The property value or default value, if no such is defined in the base Properties.
+     * @throws ConfigurationException When the given property name does not exist within the current component and the property
+     *                                does not have a default value set, or when the property cannot be parsed as a positive Integer.
+     */
+    public int getPositiveInt(Property<Integer> property, boolean strict) {
```

Comment:
I think it is better to have a separate overload
`public int getNonNegativeInt(Property<Integer> property)`
instead of passing strict.

Reply:
Done.